src/studio/src/designer/frontend/shared/features/dataModelling/DataModelling.tsx

```diff
@@ -122,16 +122,23 @@ function DataModelling({
     dispatch(DataModelsMetadataActions.getDataModelsMetadata());
   };
 
+  const loadingIndicator = (
+    <AltinnSpinner
+      spinnerText={getLanguageFromKey('general.loading', language)}
+    />
+  );
   return (
     <SchemaEditor
       language={language}
       schema={jsonSchema}
       onSaveSchema={handleSaveSchema}
       name={selectedOption?.label}
       LoadingComponent={
-        <AltinnSpinner
-          spinnerText={getLanguageFromKey('general.loading', language)}
-        />
+        metadataLoadingState === LoadingState.LoadingModels ? (
+          loadingIndicator
+        ) : (
+          <></>
```

Comment:
Prefer null over empty element
```suggestion
          null
```